### PR TITLE
beneficiaryTokenCount takes BBH into account.

### PR DESCRIPTION
**Pull request by mejango**
# Description

beneficiaryTokenCount takes BBH into account. 
multi terminal size is too big now, needs love.

## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [ ] Tests are covering the new feature
- [ ] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [ ] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [ ] I have run the test locally (and they pass)
- [ ] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:
- Directly:

- Indirectly:

### CODE REVIEW

File: src/JBMultiTerminal.sol [hunk 1]
```diff
@@ -667,8 +665,14 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
         override
         returns (uint256 beneficiaryTokenCount)
     {
+        // Get a reference to the project's tokens.
+        IJBTokens tokens = _controllerOf(projectId).TOKENS();
+
+        // Get a reference to the beneficiary's balance before the payment.
+        uint256 beneficiaryBalanceBefore = tokens.totalBalanceOf({holder: beneficiary, projectId: projectId});
```
Comment (simplemachine92) [inline src/JBMultiTerminal.sol:672]: wondering if there's any issues with including credit balances here, but, probably okay right?

File: src/JBMultiTerminal.sol [hunk 2]
```diff
@@ -678,6 +682,17 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
             metadata: metadata
         });
 
+        // Get a reference to the beneficiary's balance after the payment.
+        uint256 beneficiaryBalanceAfter = tokens.totalBalanceOf({holder: beneficiary, projectId: projectId});
+
+        // Set the beneficiary token count.
+        if (beneficiaryBalanceAfter < beneficiaryBalanceBefore) {
```
Comment (simplemachine92) [inline src/JBMultiTerminal.sol:689]: shouldn't this be an equivalence check?

**Comment by simplemachine92** [comment]
lgtm besides what I commented, will wait for @xBA5ED opinion before updating tests.

File: src/JBMultiTerminal.sol (hunk 1 above)
Comment (mejango) [reply src/JBMultiTerminal.sol:672]: def ok

**Comment by mejango** [comment]
> lgtm besides what I commented, will wait for @xBA5ED opinion before updating tests.

yep. also tests plz :]

File: src/JBMultiTerminal.sol (hunk 2 above)
Comment (simplemachine92) [reply src/JBMultiTerminal.sol:689]: thoughts on this?

Comment (mejango) [reply src/JBMultiTerminal.sol:689]: yes! great. <=

Comment (mejango) [reply src/JBMultiTerminal.sol]: added! great call

File: src/JBMultiTerminal.sol [hunk 3]
```diff
@@ -678,6 +682,17 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
...
+        uint256 beneficiaryBalanceAfter = tokens.totalBalanceOf({holder: beneficiary, projectId: projectId});
+
+        // Set the beneficiary token count.
+        if (beneficiaryBalanceAfter <= beneficiaryBalanceBefore) {
+            beneficiaryTokenCount = 0;
+        }
+        else {
+            beneficiaryTokenCount = beneficiaryBalanceAfter - beneficiaryBalanceBefore;
+        }
+
```
Comment (xBA5ED) [inline src/JBMultiTerminal.sol:695]: We can simplify this code and the statement below this into something like
```
if (minReturnedTokens != 0) {
       // Get a reference to the beneficiary's balance after the payment.
        uint256 beneficiaryBalanceAfter = tokens.totalBalanceOf({holder: beneficiary, projectId: projectId});
        
        // The token count for the beneficiary must be greater than or equal to the specified minimum.
        if ( beneficiaryBalanceBefore >= beneficiaryBalanceAfter || beneficiaryBalanceAfter - beneficiaryBalanceBefore < minReturnedTokens) {
            revert JBMultiTerminal_UnderMinReturnedTokens(beneficiaryTokenCount, minReturnedTokens);
      } 
}

Comment (mejango) [reply src/JBMultiTerminal.sol:691]: we still gotta return `beneficiaryTokenCount` tho

**Review by simplemachine92 — APPROVED** [review]

**Pull request merged**